File: 2021/1/1.py
#Testing data
with open('test.txt', 'r') as f:
    test_measurements = []
    for line in f:
        test_measurements.append(int(line.strip()))

#Actual data
with open('input.txt', 'r') as f:
    measurements = []
    for line in f:
        measurements.append(int(line.strip()))

def one_data_point_delta(measurements):
    depth_delta = {'increase': 0, 'decrease': 0}
    current_depth = 0
    for measurement in measurements:
        if current_depth == 0:
            current_depth = measurement
            continue
        elif measurement > current_depth:
            depth_delta['increase'] += 1
            current_depth = measurement
        elif measurement < current_depth:
            depth_delta['decrease'] += 1
            current_depth = measurement
    return depth_delta

def tres_data_point_delta(measurements):
    window_sums = [sum(measurements[i - 2: i + 1]) for i in range(2, len(measurements))]
    return one_data_point_delta(window_sums)

print(tres_data_point_delta(measurements))

File: 2021/1/1.js
const fs = require('fs');

// Reading testing data
const testMeasurements = fs.readFileSync('test.txt', 'utf8')
    .split('\n')
    .map(line => parseInt(line));

// Reading actual data
const measurements = fs.readFileSync('input.txt', 'utf8')
    .split('\n')
    .map(line => parseInt(line));

function oneDataPointDelta(measurements) {
    let depthDelta = { increase: 0, decrease: 0 };
    let currentDepth = 0;

    for (let i = 0; i < measurements.length; i++) {
        const measurement = measurements[i];

        if (currentDepth === 0) {
            currentDepth = measurement;
            continue;
        } else if (measurement > currentDepth) {
            depthDelta.increase++;
            currentDepth = measurement;
        } else if (measurement < currentDepth) {
            depthDelta.decrease++;
            currentDepth = measurement;
        }
    }

    return depthDelta;
}

function tresDataPointDelta(measurements) {
    const windowSums = [];

    for (let i = 2; i < measurements.length; i++) {
        windowSums.push(measurements[i - 2] + measurements[i - 1] + measurements[i]);
    }

    return oneDataPointDelta(windowSums);
}

console.log(tresDataPointDelta(measurements));
